src/Components/HistoricalDataChart/Util.js

```diff
@@ -13,7 +13,7 @@ export const formatHistoricalData = (responseData, dateRange) =>{
 
     let mapResponseData = new Map();
 
-    responseData.forEach((responseValue) => {
+    responseData.reverse().forEach((responseValue) => {
         mapResponseData.set(new Date(responseValue.report_date).toDateString(), responseValue);
     });
```

Comment:
```suggestion
  let latestResponseData = responseData[0];
    if (latestResponseData) {
        mapResponseData.set(new Date(latestResponseData.report_date).toDateString(), latestResponseData);
    }
```

Reply:
@r14chandra  -History records may include points for multiple days and not just for single day.

Reply:
yes my bad 😓

Reply:
updated.